ffi/capi/dart/package/lib/src/ICU4XAnyCalendarKind.dart

```diff
@@ -0,0 +1,164 @@
+  Ethiopian._(5),
+
+  /// The kind of an Ethiopian calendar, with Amete Alem era
+  EthiopianAmeteAlem._(6),
+
+  /// The kind of a Indian calendar
+  Indian._(7),
+
+  /// The kind of a Coptic calendar
+  Coptic._(8),
+
+  /// The kind of a Dangi calendar
+  Dangi._(9),
+
+  /// The kind of a Chinese calendar
+  Chinese._(10),
+
+  /// The kind of a Hebrew calendar
+  Hebrew._(11),
+
+  /// The kind of a Islamic civil calendar
+  IslamicCivil._(12),
+
+  /// The kind of a Islamic observational calendar
+  IslamicObservational._(13),
+
+  /// The kind of a Islamic tabular calendar
+  IslamicTabular._(14),
+
+  /// The kind of a Islamic Umm al-Qura calendar
+  IslamicUmmAlQura._(15),
+
+  /// The kind of a Persian calendar
+  Persian._(16),
+
+  /// The kind of a Roc calendar
+  Roc._(17);
+
+  const ICU4XAnyCalendarKind._(this._id);
+
+  final int _id;
+
+  /// Read the calendar type off of the -u-ca- extension on a locale.
+  ///
+  /// Errors if there is no calendar on the locale or if the locale's calendar
+  /// is not known or supported.
+  ///
+  /// See the [Rust documentation for `get_for_locale`](https://docs.rs/icu/latest/icu/calendar/enum.AnyCalendarKind.html#method.get_for_locale) for more information.
+  factory ICU4XAnyCalendarKind.getForLocale(ICU4XLocale locale) {
+    final result = _getForLocaleFfi(ICU4XLocaleAsFfi(locale));
+    return result.isOk
+        ? ICU4XAnyCalendarKindFromFfi(result.union.ok)
+        : throw VoidError();
+  }
+  static late final _getForLocaleFfi = capi<
+          ffi.NativeFunction<
+              ResultICU4XAnyCalendarKindUnionVoid Function(
+                  ICU4XLocaleFfi)>>('ICU4XAnyCalendarKind_get_for_locale')
+      .asFunction<
+          ResultICU4XAnyCalendarKindUnionVoid Function(ICU4XLocaleFfi)>();
+
+  /// Obtain the calendar type given a BCP-47 -u-ca- extension string.
+  ///
+  /// Errors if the calendar is not known or supported.
+  ///
+  /// See the [Rust documentation for `get_for_bcp47_value`](https://docs.rs/icu/latest/icu/calendar/enum.AnyCalendarKind.html#method.get_for_bcp47_value) for more information.
+  factory ICU4XAnyCalendarKind.getForBcp47(String s) {
+    final alloc = allocators.Arena();
+
+    final sList = Utf8Encoder().convert(s);
+    final sBytes = alloc.call<ffi.Char>(sList.length);
+    sBytes.cast<ffi.Uint8>().asTypedList(sList.length).setAll(0, sList);
+
+    final result = _getForBcp47Ffi(sBytes.cast(), sList.length);
+    alloc.releaseAll();
+    return result.isOk
+        ? ICU4XAnyCalendarKindFromFfi(result.union.ok)
+        : throw VoidError();
+  }
+  static late final _getForBcp47Ffi = capi<
+          ffi.NativeFunction<
+              ResultICU4XAnyCalendarKindUnionVoid Function(
+                  ffi.Pointer<ffi.Char>,
+                  ffi.Size)>>('ICU4XAnyCalendarKind_get_for_bcp47')
+      .asFunction<
+          ResultICU4XAnyCalendarKindUnionVoid Function(
+              ffi.Pointer<ffi.Char>, int)>();
```

Comment:
Thought: Wondering whether these capi type declarations and mapping to the ICU4X C functions should be in a common place, and then these .dart files just invoke the functions

Reply:
I think having them distributed is more manageable